lib/fog/azurerm/requests/network/create_virtual_network.rb

```diff
@@ -19,59 +19,55 @@ def create_virtual_network(resource_group, name, location, dns_list, subnet_addr
 
         private
 
-        def define_vnet_object(location, network_name, network_address_list, dns_list, subnet_address_list)
+        def define_vnet_object(location, address_prefixes, dns_servers, subnets)
           virtual_network = Azure::ARM::Network::Models::VirtualNetwork.new
           virtual_network.location = location
           virtual_network_properties = Azure::ARM::Network::Models::VirtualNetworkPropertiesFormat.new
 
-          if network_address_list.nil?
+          if address_prefixes.nil? || !address_prefixes.any?
             address_space = Azure::ARM::Network::Models::AddressSpace.new
             address_space.address_prefixes = ['10.2.0.0/16']
```

Comment:
default value should be either defined in constants file or as constant on top of the file